### PR TITLE
Force Docker to use our commands, not any defined ENTRYPOINT

### LINKED ISSUE #435: Images with ENTRYPOINT defined don't work as apps
Report originally from Slack by @melkorm. Images with an `ENTRYPOINT` defined don't play nicely with Dusty's attempts to register the command file as a `COMMAND`, since this'll end up passing the command file to an entrypoint binary that has no idea what to do with it.

This app spec is a workaround to the problem with an image (in this case, `berngp/docker-zabbix`) which has an ENTRYPOINT defined. If you remove the `compose: entrypoint` key, you will see the error. This key is a workaround for now.

```yaml
image: berngp/docker-zabbix

host_forwarding:
  - host_name: local.zabbix.com
    host_port: 80
    container_port: 80

commands:
  always:
    - /bin/docker-zabbix run

compose:
  entrypoint: []
```

I think the solution here is to enforce the `commands` pattern for apps and always blow away the entrypoint if it is defined.

### PR DESCRIPTION
**Pull request by thieman**
@paetling Fixes #435. I confirmed that @melkorm's Zabbix example works fine with this fix without having the workaround explicitly defined.

### CODE REVIEW

File: docs/changelog.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Changelog
 
 ## 0.4.0 (In Progress)
- * BREAKING CHANGE: Dusty now cleans up volumes when removing containers
+ * BREAKING CHANGE: Dusty now removes data volumes when removing containers. Containers needing to persist data should use volume mounts inside of `/persist` in the boot2docker VM.
```
Comment (paetling) [inline docs/changelog.md]: where is this in your code?

Comment (thieman) [inline docs/changelog.md]: I'm touching up the changelog message that was already there. It's a drive-by.

File: docs/changelog.md [hunk 2]
```diff
@@ -9,6 +9,7 @@
  * NEW: `dusty logs` now supports a `-t` option to show timestamps on the logging output
 
  * FIXED: `dusty upgrade` will now successfully upgrade Dusty from a RC version in all cases
+ * FIXED: Images with an `ENTRYPOINT` defined now work correctly when used as the basis for a Dusty app
```
Comment (paetling) [inline docs/changelog.md]: We should add to the docs that ENTRYPOINTS in Dockerfiles are ignored

Comment (thieman) [inline docs/changelog.md]: Smart
